google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClient.java

```diff
@@ -775,6 +789,175 @@ public void awaitReplication(String tableId) {
+          @Override
+          public List<String> apply(List<com.google.bigtable.admin.v2.Backup> protos) {
+            List<String> results = Lists.newArrayListWithCapacity(protos.size());
+            for (com.google.bigtable.admin.v2.Backup proto : protos) {
+              results.add(NameUtil.extractBackupIdFromBackupName(proto.getName()));
+            }
+            return results;
+          }
+        },
+        MoreExecutors.directExecutor());
+  }
+
+  public void deleteBackup(String clusterId, String backupId) {
+    ApiExceptions.callAndTranslateApiException(deleteBackupAsync(clusterId, backupId));
+  }
+
+  public ApiFuture<Void> deleteBackupAsync(String clusterId, String backupId) {
+    DeleteBackupRequest request =
+        DeleteBackupRequest.newBuilder()
+            .setName(NameUtil.formatBackupName(projectId, instanceId, clusterId, backupId))
+            .build();
+
+    return transformToVoid(this.stub.deleteBackupCallable().futureCall(request));
+  }
+
+  public Backup updateBackup(UpdateBackupRequest request) {
+    return ApiExceptions.callAndTranslateApiException(updateBackupAsync(request));
+  }
+
+  public ApiFuture<Backup> updateBackupAsync(UpdateBackupRequest request) {
+    return ApiFutures.transform(
+        stub.updateBackupCallable().futureCall(request.toProto(projectId, instanceId)),
+        new ApiFunction<com.google.bigtable.admin.v2.Backup, Backup>() {
+          @Override
+          public Backup apply(com.google.bigtable.admin.v2.Backup proto) {
+            return Backup.fromProto(proto);
+          }
+        },
+        MoreExecutors.directExecutor());
+  }
+
+  public RestoredTableResult restoreTable(RestoreTableRequest request)
+      throws ExecutionException, InterruptedException {
+    return ApiExceptions.callAndTranslateApiException(restoreTableAsync(request));
+  }
+
+  public ApiFuture<RestoredTableResult> restoreTableAsync(RestoreTableRequest request) {
+    final OperationFuture<com.google.bigtable.admin.v2.Table, RestoreTableMetadata> future =
+        this.stub
+            .restoreTableOperationCallable()
+            .futureCall(request.toProto(projectId, instanceId));
+    return ApiFutures.transformAsync(
+        future,
+        new ApiAsyncFunction<com.google.bigtable.admin.v2.Table, RestoredTableResult>() {
+          @Override
+          public ApiFuture<RestoredTableResult> apply(com.google.bigtable.admin.v2.Table table)
+              throws Exception {
+            return ApiFutures.immediateFuture(
+                // When apply is called, the future should have been resolved and it's safe to
+                // pull the metadata.
+                new RestoredTableResult(
+                    Table.fromProto(table),
+                    future.getMetadata().get().getOptimizeTableOperationName()));
+          }
+        },
+        MoreExecutors.directExecutor());
+  }
+
+  public Void awaitOptimizeRestoredTable(OptimizeRestoredTableOperationToken token)
```

Comment:
can this be the primitive void?

Reply:
Unfortunately it has to be Void class.

Reply:
why?

Reply:
If I change it to void, the complier will complain. I think Void is used for AwaitReplication as well.

Reply:
updated the method signature to void